### DISCUSSION TITLE
Error: No module named 'requests'

### DISCUSSION THREAD

**Discussion by AIZeplin** [oobabooga/text-generation-webui#1494] [Q&A]
Hi, after finally installing all the different (I think) requirements (the basic windows installer file does NOT install everything that's needed, far from it), I'm finally trying to run the actual server.py file.

But when I try to do it, I get the following error thrown back at me. I've tried searching for answers, checked out Reddit, etc., but I can't seem to find a mention of the problem/an answer. Any clue what's going wrong here?

(textgen) C:\AI\oobabooga_windows\text-generation-webui>python server.py Traceback (most recent call last): File "C:\AI\oobabooga_windows\text-generation-webui\server.py", line 2, in <module> import requests ModuleNotFoundError: No module named 'requests'

**Comment by user76765** [comment]
cd into your text-generation-webui
python -m pip install -r ./requirements.txt

**Reply by dexterfitch** [reply]
Thanks, I was following a guide that skipped that step and was stuck here too, much appreciated!

**Comment by OmidH** [comment]
I am running into the same issue on my mac (**Sonoma**), while trying to run a clean (clone) install:

```
❯ ./start_macos.sh


*******************************************************************
* WARNING: You haven't downloaded any model yet.
* Once the web UI launches, head over to the "Model" tab and download one.
*******************************************************************


Traceback (most recent call last):
  File "/Users/omid/Workspace/ai/_tools_/text-generation-webui/server.py", line 5, in <module>
    from modules.block_requests import OpenMonkeyPatch, RequestBlocker
  File "/Users/omid/Workspace/ai/_tools_/text-generation-webui/modules/block_requests.py", line 4, in <module>
    import requests
ModuleNotFoundError: No module named 'requests'
```

**Comment by SteveC** [comment]
same problem, still have the issue after installing requirements on macos

**Reply by BeefBaby** [reply]
I do too.

**Comment by Bluenochian** [comment]
I kept experiencing the same issue repeatedly, even after the requirements had finished downloading and displayed 'Requirement already satisfied' (I tried it several more times to be sure). Regardless of my efforts and hours of research, the problem persisted. And no matter what I do, the issue is still there

I even did some extra steps for that which is:

> - [1 ] Downloaded and installed Anaconda manually checking the make a PATH option.
> - [2 ] Downloaded and installed the latest version of python again checking the make a PATH option.
> - [3 ] Lastly I came back to the original folder and opened the terminal and used python -m pip install -r ./requirements.txt again.

Still the same... Any help is appreciated!

**Reply by Bluenochian** [reply]
I fixed it by deleting everything and doing a fresh install, and voila, it's fixed!

It was clearly related to the steps I followed earlier (for anyone who might need it in the future):

> [1] Manually downloaded and installed Anaconda, ensuring to check the 'Add to PATH' option.
> [2] Downloaded and installed the latest version of Python, again checking the 'Add to PATH' option.
